src/ui/index.ts

```diff
@@ -102,27 +104,79 @@ export function getNodeId({ attributes }: UiNode) {
 }
 
 /**
- * Filters nodes by their groups.
+ * Return the node input attribute type
+ * In <input> elements we have a variety of types, such as text, password, email, etc.
+ * When the attribute is null or the `type` attribute is not present, we assume it has no defined type.
+ * @param attr
+ * @returns type of node
+ */
+export const getNodeInputType = (attr: UiNodeAttributes): string =>
+  attr && "type" in attr ? attr.type : ""
+
+export type FilterNodesByGroups = {
+  nodes: Array<UiNode>
+  groups?: Array<UiNodeGroupEnum | string> | UiNodeGroupEnum | string
+  withoutDefaultGroup?: boolean
+  attributes?:
+    | Array<UiNodeInputAttributesTypeEnum | string>
+    | UiNodeInputAttributesTypeEnum
+    | string
+  withoutDefaultAttributes?: boolean
+  excludeAttributes?:
+    | Array<UiNodeInputAttributesTypeEnum | string>
+    | UiNodeInputAttributesTypeEnum
+    | string
+}
+
+/**
+ * Filters nodes by their groups and attributes.
  *
  * Will always add default nodes unless `withoutDefaultGroup` is true.
- *
+ * Will always add default attributes unless `withoutDefaultAttributes` is true.
  * @param nodes
  * @param groups
  * @param withoutDefaultGroup
+ * @param attributes
+ * @param withoutDefaultAttributes
```

Comment:
These are not really top-level args to the function, as the function takes just one object as arg.